plugins/discourse-post-voting/spec/serializers/post_voting/topic_list_item_serializer_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+describe PostVoting::TopicListItemSerializerExtension do
+  fab!(:user)
+  fab!(:topic) { Fabricate(:topic, subtype: Topic::POST_VOTING_SUBTYPE) }
+  fab!(:post) { Fabricate(:post, topic: topic, qa_vote_count: 3) }
+
+  before do
+    SiteSetting.post_voting_enabled = true
+    topic.update!(archetype: Archetype.default)
+  end
+
+  context "when post voting is disabled" do
+    before { SiteSetting.post_voting_serialize_topic_votes_count = false }
+
+    it "does not include vote count in serialized output" do
+      json = described_class.new(topic, scope: Guardian.new(user)).as_json
+      expect(json[:topic_list_item][:topic_vote_count]).to be_nil
+    end
+  end
+
+  context "when post voting is enabled" do
+    before { SiteSetting.post_voting_serialize_topic_votes_count = true }
+
+    it "includes topic_vote_count when mode is total" do
```

Comment:
Do we want to add another test that ensures deleted posts aren't being included in the total? 🤔